astropy/io/fits/tests/test_core.py

```diff
@@ -593,6 +595,48 @@ def test_open_nonexistent(self):
             assert os.path.exists(self.temp('foobar.fits'))
             os.remove(self.temp('foobar.fits'))
 
+    @pytest.mark.skipif(six.PY2,
```

Comment:
I'll leave this decision up to someone else but I'd be happy to add version-specific logic if it makes it easier to integrate the changes sooner.